### PR TITLE
Attempt to fix rendering of translated fields

### LINKED ISSUE #1849: Second level translated fields are stored but not rendered back on read
### Versions 
App: 7.6.2
Api: 2.2.1

### How to  reproduce
Create collection structure with second level translated fields in directus.

[schema diagram](https://dbdiagram.io/d/5d2c4d6bced98361d6dca889)

1. Create new article
2. Add new section
3. Add translated value *section_name*
4. Save. Observe that save is successful
5. Read the article again. Notice that the translated fields are empty

### PR DESCRIPTION
**Pull request by oliverpool**
Fix #1849 

The `valuesByLang` was:

```
[
  {
    description: "some description",
    language: { language: "en" }
  }
]
```

but it should be:

```
[
  {
    description: "some description",
    language: "en"
  }
]
```

### CODE REVIEW

**Comment by oliverpool** [comment]
It seems to work on https://deploy-preview-1879--directus-nightly.netlify.com/#/collections/concessions/5 (where I created a related translation)

**Comment by rijkvanzanten** [comment]
LGTM! Thanks @oliverpool

**Comment by Kinzi** [comment]
This fix breaks the translation interface for me. The fields in the translation collection are not populated. If I revert the changes everything works again.

Ideas?

**Comment by rijkvanzanten** [comment]
Strange, I tried the PR and it did work on my end.. I'll revisit it as soon as I can

**Comment by oliverpool** [comment]
@Kinzi how is your database ? Do you have a `en`column or a complete translation table with foreign keys?

**Comment by Kinzi** [comment]
@oliverpool Not quite sure what you mean - but I think the latter:
Bildschirmfoto 2019-07-31 um 11 16 43
Bildschirmfoto 2019-07-31 um 11 16 33

**Comment by oliverpool** [comment]
@Kinzi I guess it depends on the GET request.

On the demo-api, the item is requested with the field: `translations.*.*` (which makes the `language` not only an id, but a full blown object).
I guess that your GET call does not contain this (but maybe something like `translations.*`)

https://demo-api.directus.app/_/items/movies/167?fields=id%2Ctranslations.*

---

Supporting both cases should be feasible.

File: src/interfaces/translation/input.vue [hunk 1]
```diff
@@ -137,7 +141,11 @@ export default {
             ];
       } else {
         newValue = this.value.map(translation => {
-          if (translation[this.options.translationLanguageField] === this.activeLanguage) {
```
Comment (oliverpool) [inline src/interfaces/translation/input.vue]: This one should be easy:
```js
let language = translation[this.options.translationLanguageField]
if (
language === this.activeLanguage ||
language[this.options.languagesPrimaryKeyField] === this.activeLanguage
){
) {

File: src/interfaces/translation/input.vue [hunk 2]
```diff
@@ -78,7 +78,11 @@ export default {
     },
     valuesByLang() {
       if (!this.value) return {};
-      return _.keyBy(this.value, this.options.translationLanguageField);
```
Comment (oliverpool) [inline src/interfaces/translation/input.vue]: A little bit harder:

```js
let firstKey = this.options.translationLanguageField
let secondKey = this.options.languagesPrimaryKeyField
return _(this.value).map((v)=> {
  v[firstKey] = v[firstKey][secondKey] || v[firstKey]
  return v
}).keyBy(this.options.translationLanguageField)
```

**Comment by oliverpool** [comment]
@Kinzi does https://github.com/directus/app/pull/1914 fixes it for you?

**Comment by Kinzi** [comment]
@oliverpool Looks good thanks! 🙏😉

**Pull request merged**